pkg/backend/snapshot.go

```diff
@@ -662,8 +662,13 @@ func (sm *SnapshotManager) snap() *deploy.Snapshot {
 		secretsManager = sm.baseSnapshot.SecretsManager
 	}
 
+	var metadata deploy.SnapshotMetadata
+	if sm.baseSnapshot != nil {
```

Comment:
We always keep the old metadata?

Reply:
Discussed offline -- we think this makes sense (it's just a bit weirder without the context from #17291)